test/functional/services/remote/poll_for_log_entry.ts

```diff
@@ -19,48 +19,90 @@
 
 import { WebDriver, logging } from 'selenium-webdriver';
 import * as Rx from 'rxjs';
-import { mergeMap, catchError, repeatWhen, mergeMapTo, delay } from 'rxjs/operators';
+import { mergeMap, catchError, mergeMapTo, delay, first } from 'rxjs/operators';
 
 /**
  * Create an observable that emits log entries representing the calls to log messages
  * available for a specific logger.
  */
-export function pollForLogEntry$(driver: WebDriver, type: string, ms: number) {
+export function pollForLogEntry$(
+  driver: WebDriver,
+  type: string,
+  ms: number,
+  stop$: Rx.Observable<undefined>
+) {
   const logCtrl = driver.manage().logs();
+  const poll$ = new Rx.BehaviorSubject(undefined);
 
-  // setup log polling
-  return Rx.defer(async () => await logCtrl.get(type)).pipe(
-    // filter and flatten list of entries
-    mergeMap(entries =>
-      entries.filter(entry => {
-        // ignore react devtools
-        if (entry.message.includes('Download the React DevTools')) {
-          return false;
-        }
+  const FINAL_MSG = '@@final@@';
 
-        // down-level inline script errors
-        if (entry.message.includes('Refused to execute inline script')) {
-          entry.level = logging.getLevel('INFO');
-        }
+  return new Rx.Observable<logging.Entry>(subscriber => {
+    subscriber.add(
+      stop$.pipe(first()).subscribe(() => {
+        driver
+          .executeScript(
+            `
+              if (window.flushCoverageToLog) {
+                window.flushCoverageToLog();
+              }
 
-        return true;
+              console.log(${JSON.stringify(FINAL_MSG)})
+            `
+          )
+          .catch(error => subscriber.error(error));
       })
-    ),
+    );
```

Comment:
should we wrap it with `if(collectCoverage){...}` to skip when we don't run tests with coverage?

Reply:
I don't think we need to